test/src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -321,6 +321,10 @@ public Jenkins getInstance() {
      * @throws Throwable if setup fails (which will disable {@code after}
      */
     public void before() throws Throwable {
+        if (Thread.interrupted()) { // JENKINS-30395
+            System.err.println("was interrupted before start");
```

Comment:
actually loggers and direct writing to system out/err use different buffering strategies and you can get interleaved logs :-(